test.ts

```diff
@@ -40,136 +72,718 @@ describe('does parsing', () => {
+      quiet: true,
+    });
   });
 
   it("boolean with prefixed 'no'", () => {
-
+    const [options] = parse(
+      ['--no-stars', '--no-quiet'],
+      {
+        'no-stars': {
+          alias: 'y',
+          desc: 'oh I believe in yesterday.',
+          type: 'boolean'
+        },
+        'no-quiet': {
+          alias: 'q',
+          desc: 'the quiet one.',
+          type: 'boolean'
+        }
+      }
+    );
+    expect(options).toEqual({
+      'no-stars': undefined,
+      'no-quiet': undefined
+    });
   });
 
   it('boolean with rhs true', () => {
-
+    const [options] = parse(
+      ['--stars=true', '--moon=true'],
+      {
+        'stars': {
+          alias: 'y',
+          desc: 'oh I believe in yesterday.',
+          type: 'boolean'
+        },
+        'moon': {
+          alias: 'q',
+          desc: 'the quiet one.',
+          type: 'boolean'
+        }
+      }
+    );
+    expect(options).toEqual({
+      'stars': true,
+      'moon': true
+    });
   });
 
   it('boolean with rhs false', () => {
-
+    const [options] = parse(
+      ['--stars=false', '--moon=false'],
+      {
+        'stars': {
+          alias: 'y',
+          desc: 'oh I believe in yesterday.',
+          type: 'boolean'
+        },
+        'moon': {
+          alias: 'q',
+          desc: 'the quiet one.',
+          type: 'boolean'
+        }
+      }
+    );
+    expect(options).toEqual({
+      'stars': false,
+      'moon': false
+    });
   });
 
+  // TODO: it's ok ?
   it("boolean with any value that not equal to 'false'", () => {
-
+    const [options] = parse(
+      ['--stars=true', '--moon=true'],
+      {
+        'stars': {
+          alias: 'y',
+          desc: 'oh I believe in yesterday.',
+          type: 'boolean'
+        },
+        'moon': {
+          alias: 'q',
+          desc: 'the quiet one.',
+          type: 'boolean'
+        }
+      }
+    );
+    expect(options).toEqual({
+      'stars': true,
+      'moon': true
+    });
   });
 
   it('number with rhs hexadecimal value', () => {
-
+    const [options] = parse(
+      ['--starfish=1314', '--hippocampus=1024'],
+      {
+        'starfish': {
+          alias: 'y',
+          desc: 'oh I believe in yesterday.',
+          type: 'number'
+        },
+        'hippocampus': {
+          alias: 'q',
+          desc: 'the quiet one.',
+          type: 'number'
+        }
+      }
+    );
+    expect(options).toEqual({
+      'starfish': 0x522,
+      'hippocampus': 0x400
+    });
   });
 
   it('number with rhs decimal value', () => {
-
+    const [options] = parse(
+      ['--starfish=1314', '--hippocampus=1024'],
+      {
+        'starfish': {
+          alias: 'y',
+          desc: 'oh I believe in yesterday.',
+          type: 'number'
+        },
+        'hippocampus': {
+          alias: 'q',
+          desc: 'the quiet one.',
+          type: 'number'
+        }
+      }
+    );
+    expect(options).toEqual({
+      'starfish': 0o2442,
```

Comment:
decimal is ten based. I don't what you to do math arithmetic operations

Reply:
It's a mistake.